src/Faker/Provider/HtmlLorem.php

```diff
@@ -0,0 +1,261 @@
+    const P_TAG = "p";
+    const A_TAG = "a";
+    const SPAN_TAG = "span";
+    const TABLE_TAG = "table";
+    const THEAD_TAG = "thead";
+    const TBODY_TAG = "tbody";
+    const TR_TAG = "tr";
+    const TD_TAG = "td";
+    const TH_TAG = "th";
+    const UL_TAG = "ul";
+    const LI_TAG = "li";
+    const H_TAG = "h";
+    const B_TAG = "b";
+    const I_TAG = "i";
+    const TITLE_TAG = "title";
+    const FORM_TAG = "form";
+    const INPUT_TAG = "input";
+    const LABEL_TAG = "label";
+
+    /**
+     * @return string
+     */
+    public static function randomHtml($maxDepth = 4, $maxWidth = 4)
+    {
+        $document = new \DOMDocument();
+
+        $head = $document->createElement("head");
+        static::addRandomTitle($head);
+
+        $body = $document->createElement("body");
+        static::addLoginForm($body);
+        static::addRandomSubTree($body, $maxDepth, $maxWidth);
+
+        $html = $document->createElement("html");
+        $html->appendChild($head);
+        $html->appendChild($body);
+
+        $document->appendChild($html);
+        return $document->saveHTML();
+    }
+
+    private static function addRandomSubTree(\DOMElement $root, $maxDepth, $maxWidth)
+    {
+        $maxDepth--;
+        if ($maxDepth <= 0) {
+            return $root;
+        }
+
+        $siblings = mt_rand(1, $maxWidth);
+        for ($i = 0; $i < $siblings; $i++) {
+            if ($maxDepth == 1) {
+                static::addRandomLeaf($root);
+            } else {
+                $sibling = $root->ownerDocument->createElement("div");
+                $root->appendChild($sibling);
+                static::addRandomAttribute($sibling);
+                static::addRandomSubTree($sibling, mt_rand(0, $maxDepth), $maxWidth);
+            }
+        };
+        return $root;
+    }
+
+    private static function addRandomLeaf(\DOMElement $node)
+    {
+        $rand = mt_rand(1, 10);
+        switch($rand){
+            case 1:
+                static::addRandomP($node);
+                break;
+            case 2:
+                static::addRandomA($node);
+                break;
+            case 3:
+                static::addRandomSpan($node);
+                break;
+            case 4:
+                static::addRandomUL($node);
+                break;
+            case 5:
+                static::addRandomH($node);
+                break;
+            case 6:
+                static::addRandomB($node);
+                break;
+            case 7:
+                static::addRandomI($node);
+                break;
+            case 8:
+                static::addRandomTable($node);
+                break;
+            default:
+                static::addRandomText($node);
+                break;
+        }
+    }
+
+    private static function addRandomAttribute(\DOMElement $node)
+    {
+        $rand = mt_rand(1, 2);
+        switch ($rand) {
+            case 1:
+                $node->setAttribute("class", Lorem::word());
```

Comment:
how would i use $this in a static method? The mix of static / non-static in the faker is slightly confusing.